ts/src/program/namespace/account.ts

```diff
@@ -345,6 +343,13 @@ export class AccountClient<
   ): Promise<PublicKey> {
     return await pubkeyUtil.associated(this._programId, ...args);
   }
+
+  async getAccountInfo(address: Address, commitment?: Commitment) {
```

Comment:
I think this `async` can be removed?

Reply:
Interesting, `connection.getAccountInfo` return `Promise`, but no `await` inside the function.
Q: function return promise or result? :slightly_smiling_face:

Reply:
yea you dont need async if you just return sth and theres no await inside the function. if it returns a promise the caller can await. I get the clarity argument but wouldnt it be more idiomatic to have an explicit return type then instead of inference and `async` which implies there is an await.

Reply:
cool happy to amend that, wasn't sure what the convention was for this repo.

Would it make sense to add an eslint rule to enforce it in the future if that's the case?

Reply:
my answer was a question. not sure what the convention is myself. dont think we have one except "make prettier happy" for now. but I think that can be a discussion for a different issue

Reply:
I'd prefer to have `async getAccountInfo` with inner `await`.

Reply:
If it helps, there are subtle differences to return Vs return await besides just syntactic ones.

https://eslint.org/docs/rules/no-return-await

https://jakearchibald.com/2017/await-vs-return-vs-return-await/

Don't mind either way, but something to take into consideration perhaps?

Reply:
> I think another difference is that the V8 engine provides better stack traces with return await
so it would seem!
https://github.com/goldbergyoni/nodebestpractices/issues/737

let's do `return await` then